### DISCUSSION TITLE
Build URL without sending request

### DISCUSSION THREAD

**Discussion by epotter24G** [ecyrbe/zodios#338] [Ideas]
Let's say I want to send an email to a user that has a magic link. Today, I would just write the url by hand in the email message:
```ts
const url = `${env.API_URL}/api/auth/magicLink?someId=${someId}&token=${token}`
```
But if that endpoint is defined with zodios, it would be convenient to be able to build that URL via zodios as well.
```ts
...
const url = urlBuilder.magicLink({ queries: { someId, token }});
// url value should be the same in both instances
```

**Comment by ecyrbe** [comment]
This can be implemented with little effort.
A PR is welcomed, you can inspire yourself with the react `getKeyByPath` and `getKeyByAlias` in zodios react hooks.

This could most likely look to something like this
```ts
const url = zodiosUrlBuilder(zodiosClient).<alias>({ params, queries });
```
should need to support both path parameters and query parameters

Obviously you can even build that in userland outside of zodios in a dedicated library. Since zodios exports all the Typescript Helpers to do this and still have typesafety and auto completion

**Reply by epotter24G** [reply]
Alright, I should be able to take a look in the next couple days 🙂
